File: ios/StockAgents/Utilities/Theme.swift
import SwiftUI

/// Color helpers so the hex strings ported from the web app (`NODE`, `STAGES`,
/// `RATING_COLOR`) can be used directly in SwiftUI.
extension Color {
    /// Build a Color from a "#rrggbb" / "rrggbb" hex string. Falls back to gray.
    init(hex: String) {
        let cleaned = hex.trimmingCharacters(in: CharacterSet.alphanumerics.inverted)
        var value: UInt64 = 0
        guard Scanner(string: cleaned).scanHexInt64(&value), cleaned.count == 6 else {
            self = .gray
            return
        }
        let red = Double((value & 0xFF0000) >> 16) / 255.0
        let green = Double((value & 0x00FF00) >> 8) / 255.0
        let blue = Double(value & 0x0000FF) / 255.0
        self = Color(red: red, green: green, blue: blue)
    }
}

enum Theme {
    static let background = Color(hex: "#0f172a")
    static let surface = Color(hex: "#1e293b")
    static let surfaceMuted = Color(hex: "#0b1220")
    static let border = Color(hex: "#334155")
    static let textPrimary = Color(hex: "#e2e8f0")
    static let textMuted = Color(hex: "#94a3b8")
    static let accent = Color(hex: "#34d399")
    static let warning = Color(hex: "#fbbf24")
    static let danger = Color(hex: "#f43f5e")
    static let upColor = Color(hex: "#34d399")
    static let downColor = Color(hex: "#fb7185")
}

File: mobile-rn/src/theme/theme.ts
/**
 * Dark theme ported verbatim from the SwiftUI app (`ios/StockAgents/Utilities/
 * Theme.swift`), which in turn mirrors the web app's palette. Keeping the exact
 * hex values guarantees visual parity across SwiftUI / web / React Native.
 */
export const colors = {
  background: '#0f172a',
  surface: '#1e293b',
  surfaceMuted: '#0b1220',
  border: '#334155',
  textPrimary: '#e2e8f0',
  textMuted: '#94a3b8',
  accent: '#34d399',
  warning: '#fbbf24',
  danger: '#f43f5e',
  up: '#34d399',
  down: '#fb7185',
} as const;

export const spacing = {
  xs: 4,
  sm: 8,
  md: 12,
  lg: 16,
  xl: 24,
} as const;

export const radius = {
  sm: 10,
  md: 12,
  lg: 14,
  xl: 16,
} as const;

/**
 * Type-scale tokens. The minimum body / label size is `sm` (13px) which keeps
 * `colors.textMuted` (#94a3b8 on #0f172a) at the WCAG AA 4.5:1 threshold.
 * Reserve `xs` (11px) for captions rendered in `colors.textPrimary` only.
 */
export const fontSize = {
  xs: 11,
  sm: 13,
  md: 14,
  lg: 16,
  xl: 20,
  display: 28,
} as const;

/** Apple HIG minimum touch target. Android's 48dp recommendation is satisfied at 1x density. */
export const minHitSize = 44;

/** Add alpha to a #rrggbb hex (parity with SwiftUI `.opacity(...)`). */
export function withAlpha(hex: string, alpha: number): string {
  const a = Math.round(Math.min(1, Math.max(0, alpha)) * 255)
    .toString(16)
    .padStart(2, '0');
  return `${hex}${a}`;
}
